src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private int currentCapacity;
+    private int currentLoadFactor;
+    private Node<K, V>[] storage;
+
+    public MyHashMap() {
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        currentLoadFactor = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        storage = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeed();
+        int position = getPosition(key);
+        Node<K, V> oldNode = storage[position];
+        if (oldNode == null) {
+            storage[position] = new Node<>(key, value);
+            size++;
+        } else {
+            Node<K, V> nextNode = oldNode;
+            while (nextNode != null) {
+                if (Objects.equals(nextNode.key, key)) {
+                    nextNode.value = value;
+                    break;
+                }
+                if (nextNode.nextNode == null) {
+                    nextNode.nextNode = new Node<>(key, value);
+                    size++;
+                    return;
+                }
+                nextNode = nextNode.nextNode;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        V value = null;
+        Node<K, V> currentNode = storage[getPosition(key)];
+        if (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                value = currentNode.value;
+            } else {
```

Comment:
you don't definitely need the whole this block. Left only while and logic inside while.

Reply:
++